src/meta/Base.ts

```diff
@@ -1,15 +1,18 @@
+import { Destroyable } from '@dojo/core/Destroyable';
 import global from '@dojo/shim/global';
 import Map from '@dojo/shim/Map';
 import Set from '@dojo/shim/Set';
 import { WidgetMetaProperties } from '../interfaces';
 
-export class Base {
+export class Base extends Destroyable {
 	private _invalidate: () => void;
 	private _invalidating: number;
 	private _requiredNodes: Set<string>;
 	protected nodes: Map<string, HTMLElement>;
```

Comment:
Both the `_requiredNodes` and `nodes` should be cleared on destruction so they do not hold their references and can be freed for GC.